### PR TITLE
Role positioning

### PR DESCRIPTION
**Pull request by SnazzyPine25**

### CODE REVIEW

**Comment by SnazzyPine25** [comment]
#119

**Comment by PoVa** [comment]
Just run `rubocop` so you don't have to commit so often.

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -645,6 +646,16 @@ def hoist=(hoist)
       update_role_data(hoist: hoist)
     end
 
+    # Changes role position
+    # @param position [Integer] The value it should be changed to
+    def position=(position)
+      if position.nil? || position.is_a?(Number) && position > 1
+        update_role_data(position: position)
+      else
+        raise "Role number can't be less than or equal to 1!"
```
Comment (meew0) [inline lib/discordrb/data.rb]: Can you make it raise `ArgumentError` here?

**Comment by SnazzyPine25** [comment]
@meew0 something like `raise ArgumentError` ?

**Comment by meew0** [comment]
```rb
raise ArgumentError, "Role number can't be less than or equal to 1!"
```

**Comment by SnazzyPine25** [comment]
Alright Ill make a commit

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -645,6 +646,16 @@ def hoist=(hoist)
       update_role_data(hoist: hoist)
     end
 
+    # Changes role position
+    # @param position [Integer] The value it should be changed to
+    def position=(position)
+      if position.nil? || position.is_a?(Number) && position > 1
```
Comment (meew0) [inline lib/discordrb/data.rb]: `Number` is not defined, should be `Integer`

**Comment by meew0** [comment]
Still doesn't work, it appears you're calling `API.update_role` with the wrong number of arguments

**Comment by SnazzyPine25** [comment]
Was I supposed to include the role data?

**Comment by SnazzyPine25** [comment]
Whoops wrong button

**Comment by meew0** [comment]
You need to change `API.update_role` to also change the role position, otherwise you'll get an error when calling it.

**Comment by meew0** [comment]
During testing, I found out that apparently you can't set a single role's position over the usual update_role endpoint - you have to change multiple roles' positions at once using the [batch modify endpoint](https://discordapp.com/developers/docs/resources/guild#modify-guild-role) or the change gets dropped (because other roles are apparently in the way). If you want, you can give it another try, but in its current form the PR won't work, so I'll close it.

**Pull request closed without merging**